manifests/hash.pp

```diff
@@ -66,7 +66,8 @@
 
   file {"${name}.db":
     ensure  => $ensure,
-    require => [File[$name], Exec["generate ${name}.db"]],
+    require => File[$name],
+    notify  => Exec["generate ${name}.db"],
```

Comment:
Correct, otherwise puppet will create a zero byte file for us which is the opposite of what we want.